### ISSUE TITLE
Start USB support

### ISSUE THREAD

**Issue by WSLUser** [microsoft/WSL2-Linux-Kernel#23]
From https://github.com/rpasek/usbip-wsl2-instructions:

>  This can be included by default: 
> Navigate in menuconfig to select the USB kernel modules you'd like. These suited my needs but add more or less as you see fit:
```

Device Drivers->USB support[*]
Device Drivers->USB support->Support for Host-side USB[M]
Device Drivers->USB support->Enable USB persist by default[*]
Device Drivers->USB support->USB Modem (CDC ACM) support[M]
Device Drivers->USB support->USB Mass Storage support[M]
Device Drivers->USB support->USB/IP support[M]
Device Drivers->USB support->VHCI hcd[M]
Device Drivers->USB support->VHCI hcd->Number of ports per USB/IP virtual host controller(8)
Device Drivers->USB support->Number of USB/IP virtual host controllers(1)
Device Drivers->USB support->USB Serial Converter support[M]
Device Drivers->USB support->USB Serial Converter support->USB FTDI Single Port Serial Driver[M]
Device Drivers->USB support->USB Physical Layer drivers->NOP USB Transceiver Driver[M]
Device Drivers->Network device support->USB Network Adapters[M]
Device Drivers->Network device support->[Deselect everything you dont care about]
Device Drivers->Network device support->USB Network Adapters->Multi-purpose USB Networking Framework[M]
Device Drivers->Network device support->USB Network Adapters->CDC Ethernet support (smart devices such as cable modems)[M]
Device Drivers->Network device support->USB Network Adapters->Multi-purpose USB Networking Framework->Host for RNDIS and ActiveSync devices[M]
```

**Comment by swarren** [comment]
https://github.com/microsoft/WSL2-Linux-Kernel/pull/45 might be relevant.

**Comment by einarpersson** [comment]
Any news on this? When can we expect USB-support? 

The only* big thing I am missing from WSL2 right now is being able to use [PlatformIo](https://platformio.org/) in WSL2 for developing Arduino etc. 

*aside from a few irritating bugs

**Comment by miloshavlicek** [comment]
@sirredbeard Any news about the official support in default build? It's so missing feature.

**Comment by markeissler** [comment]
I just stumbled across this article which leverages iSCSI to support connection of USB devices...

https://github.com/jovton/USB-Storage-on-WSL2

Perhaps it could be a useful reference? The thing is it needs:

- recompile of kernel for SCSI and Fuse support
- add a `.wslconfig` file to the user's Windows directory
- call `modprobe` as needed in the target container

I'd think the first two were easy (update the distributed kernel from MS, have WSL2 drop the config file in the user's home directory). Don't know how the last one could be handled but then if you can copy files into a container to support the `vscode-server` and then also run things in that container to support vscode remote sessions, then perhaps running modprobe might be easy enough...

**Comment by WSLUser** [comment]
`recompile of kernel for SCSI and Fuse support`

FUSE is already supported. SCSI is hardware and is N/A in WSL2 until some method of utilizing the USB on Windows is created. However USBIP does work. I don't think it would be wise to bloat up the kernel with a bunch of physical drivers that are useless sorry. Use the method described in the OP. Also use of `=m` is not supported in WSL2 currently.

> add a .wslconfig file to the user's Windows directory

This will never happen or we'd have `etc/wsl.conf` already created for us.

**Comment by mtrojnar** [comment]
> However USBIP does work.

No, it does not (at least with the latest distributed kernel release build, i.e., 4.19.104-microsoft-standard).
https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi
https://docs.microsoft.com/en-us/windows/wsl/kernel-release-notes

> I don't think it would be wise to bloat up the kernel with a bunch of physical drivers that are useless sorry.

USB device drivers are only useless without USB support, which is precisely the subject of this issue.

What exactly do you mean exactly by "I don't think it would be wise to bloat up the kernel"? Would a dozen of USB device drivers substantially increase the kernel size?  Have you measured the memory footprint of those drivers?

> Also use of `=m` is not supported in WSL2 currently.

Yes, kernel module support would be nice, but it is a distinct feature.  I don't think USB support should depend on kernel module support.

**Comment by WSLUser** [comment]
Search for the WSL issue for USB support. Theres a repo that allows a specific Windows application to connect to its Linux counterpart in WSL2 that enables USB support through USB-IP. The idea of WSL2 is to be performant and low foot print. Each module enabled grows the size of the kernel. This can be acceptable depending on the module. Physical drivers are unable to be utilized by WSL2. Therefore they do nothing but increase the size of the kernel.
